WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ProductListRepository.kt

```diff
@@ -165,12 +165,18 @@ class ProductListRepository @Inject constructor(
         productFilterOptions: Map<ProductFilterOption, String> = emptyMap(),
         excludedProductIds: List<Long>? = null
     ): List<Product> {
+        // we need to pass null because FluxC crashes when an empty list is passed
+        val excludedIds = if (excludedProductIds?.isNotEmpty() == true) {
+            excludedProductIds
+        } else {
+            null
+        }
```

Comment:
np: If think it's useful, you can take advantage of the `takeIf` to also imply a nullable condition clause
```suggestion
        val excludedIds = excludedProductIds?.takeIf { it.isNotEmpty() }
```

Reply:
Thanks for the tip, that's a much cleaner approach! I made the change in 0b99bcf.